android/app/src/googleplay/java/org/fossasia/openevent/utils/map/ClusterRenderer.java

```diff
@@ -0,0 +1,94 @@
+package org.fossasia.openevent.utils.map;
+
+import android.content.Context;
+import android.os.Handler;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.maps.android.clustering.ClusterManager;
+import com.google.maps.android.clustering.view.DefaultClusterRenderer;
+
+import org.fossasia.openevent.R;
+
+/**
+ * Created by rohanagarwal94 on 4/7/17.
+ */
+public class ClusterRenderer extends DefaultClusterRenderer<MicrolocationClusterWrapper> implements ClusterManager.OnClusterItemClickListener<MicrolocationClusterWrapper>, GoogleMap.OnMapClickListener {
+
+    private Context context;
+    private GoogleMap map;
+    private MicrolocationClusterWrapper microlocationClusterWrapper;
+
+    public ClusterRenderer(Context context, GoogleMap map,
+                             ClusterManager<MicrolocationClusterWrapper> clusterManager) {
+        super(context, map, clusterManager);
+        this.context = context;
+        this.map = map;
+        setOnClusterItemClickListener(this);
+    }
+
+    @Override
+    protected void onBeforeClusterItemRendered(MicrolocationClusterWrapper item, MarkerOptions markerOptions) {
+        super.onBeforeClusterItemRendered(item, markerOptions);
+
+        markerOptions.title(item.getMicrolocation().getName());
+        if (microlocationClusterWrapper != null && item.equals(microlocationClusterWrapper)) {
+            markerOptions.icon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.color_primary));
+        } else {
+            markerOptions.icon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.dark_grey));
+        }
+    }
+
+    @Override
+    protected void onClusterItemRendered(final MicrolocationClusterWrapper clusterItem, Marker marker) {
+        super.onClusterItemRendered(clusterItem, marker);
+        clusterItem.setMarker(marker);
+   }
+
+    public void destroyContext() {
```

Comment:
Make it `detach` or `cleanUp` and make context, map and clusterManager null. The practce is to not keep strong reference to objects passed from view in classes which can stay longer in memory (Listeners)